test/jdk/javax/swing/JButton/HtmlButtonImageTest/HtmlButtonImageTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+*/
+
+/*
+ * @test
+ * @key headful
+ * @bug 8015854
+ * @requires (os.family == "mac")
+ * @summary Tests HTML image as JButton text for unwanted padding on Aqua LAF on MacOS
+ * @run main HtmlButtonImageTest
+ */
+
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.Point;
+import java.awt.Robot;
+import java.nio.file.Path;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import java.awt.Color;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+import java.io.File;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+
+public final class HtmlButtonImageTest {
+    private static JFrame frame;
+    private static Point point;
+    private static JButton button;
+    private static Path testDir;
+
+    public static final int BUTTON_HEIGHT = 37;
+    public static final int BUTTON_WIDTH = 37;
+    public static final int SQUARE_HEIGHT = 19;
+    public static final int SQUARE_WIDTH = 19;
+    public static final int PIXEL_BUFFER = 1;
+
+    public static void main(String[] args) throws Exception {
+        try {
+            UIManager.setLookAndFeel("com.apple.laf.AquaLookAndFeel");
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException("Unsupported LookAndFeel: " + e);
+        }
+
+        Robot robot = new Robot();
+        robot.setAutoDelay(2000);
+        robot.setAutoWaitForIdle(true);
+
+        // store path to source directory to locate image
+        testDir = Path.of(System.getProperty("test.classes", "."));
+
+        // generate red_square.png image to use as JButton text
+        generateImage();
+
+        SwingUtilities.invokeAndWait(() -> {
+            createAndShowGUI();
+        });
+
```

Comment:
normally it is recommended to wait for a sec after frame is visible and before start of robot events.